test/test_torch.py

```diff
@@ -6383,6 +6384,119 @@ def test_clamp(self, device):
         with self.assertRaisesRegex(RuntimeError, error_msg):
             m1.clamp_()
 
+    def test_clamp_with_tensors(self, device):
```

Comment:
Always great to see more tests!

Before this PR is landed, contact @ailzhang to fix the PyTorch/XLA build. Otherwise landing this test will break it (as it happening in the current CI). 

Additionally, this test is really big! While a lot of the tests in test_torch.py are huge and test multiple things, we'd like to cut them up into smaller, more self-contained tests. It looks like most of these blocks are separable, and you can write a helper function to generate m1 if you like. 

Is there any place you're testing backwards?

Reply:
Thanks, I have split up the new tests for clamp_with_tensors and the old ones for clamp into smaller units.

> Is there any place you're testing backwards?
The backwards behaviour is tested by test_autograd. The test definitions for the new functionality are in `common_method_invocations.py`. It tests the backward behaviour by comparing the output from the code vs the numerical approximation using small finite distances. It also compares the outputs from the CPU kernel vs CUDA.

I will reach out to with the contact to fix the XLA build.

Reply:
Just @ailzhang is sufficient. We also have a decorator, @onlyOnCPUAndCUDA that will prevent the tests from running on XLA. The XLA team can then update their behavior and remove the decorator on their own timeline.